### ISSUE TITLE
AOR - broken Reports - blank page

### ISSUE THREAD

**Issue by AdminPL** [salesagility/SuiteCRM#5073]
#### Issue
I'm trying to create report of all invoices paid by customers in last month but when i select date range to custom field I created  ( as condition ) report page is blank, everything was working good in 7.9.9, on 7.9.11 i can't do anything. Tried all repair options in system, clearing cache, etc.

![image](https://user-images.githubusercontent.com/30261902/35678026-1ed2072e-0752-11e8-870f-3938ad559a95.png)

#### Your Environment

* SuiteCRM Version used: 7.9.11
* Browser name and version  Chrome Version 51.0.2704.63 (64-bit))
* Environment name and version  MySQL, PHP 7.1
* Operating System and version Ubuntu

**Comment by e-reeley** [comment]
Hi @AdminPL 

Im sorry, I am unable to reproduce this issue following the steps that you mention here.

Is there any chance that you could fully outline the steps you take to get to this blank screen issue and also anything else that you feel links with this please?

Thanks.

**Comment by AdminPL** [comment]
I chcecked general log but found nothing inside. 

Report structure is :
Invoice number, invoice name, status, Pay date, assigned user.

Condition are:
status, pay date, assigned user.

When i leave conditions only to paid and set main group as assigned user everything works fine. Whenever i add time peroid or asigned user as condition report page is empty. 

In general log i see no problems connected with issue

Problem remains only while trying to report invoice module.

**Comment by e-reeley** [comment]
Thank you, I'll follow these outlined steps and see if i encounter a similar issue.

I will update here with what I find.

Thanks.

**Comment by e-reeley** [comment]
Im sorry, I cant seem to replicate this issue.

Have you added any custom code at all that may be interfering with the reports module or is this a fresh install that you are using?

Thanks again.

**Comment by AdminPL** [comment]
Pay date is custom filed added through system.

**Comment by AdminPL** [comment]
No direct changes in code.

**Comment by e-reeley** [comment]
Sorry, I still seem to get no issue, works fine for me. 

Thanks.

**Comment by AdminPL** [comment]
I turned on showing errors and got:
Notice: Undefined property: AORReportsDashlet::$seedBean occurred in /var/www/crm.fortecait.pl/include/Dashlets/Dashlet.php on line 246 [2018-02-02 09:05:21] display_stack_trace caller, file: /var/www/crm.fortecait.pl/include/utils.php line#: 3517
/var/www/crm.fortecait.pl/include/Dashlets/Dashlet.php[L:246](:StackTraceErrorHandler)
/var/www/crm.fortecait.pl/modules/Home/index.php[L:239](Dashlet:getHeader)
/var/www/crm.fortecait.pl/modules/Home/views/view.list.php[L:52](:include)
/var/www/crm.fortecait.pl/include/MVC/View/SugarView.php[L:199](HomeViewList:display)
/var/www/crm.fortecait.pl/include/MVC/Controller/SugarController.php[L:432](SugarView:process)
/var/www/crm.fortecait.pl/include/MVC/Controller/SugarController.php[L:375](SugarController:processView)
/var/www/crm.fortecait.pl/include/MVC/SugarApplication.php[L:105](SugarController:execute)
/var/www/crm.fortecait.pl/index.php[L:52](SugarApplication:execute)
Warning: Declaration of SugarWidgetFieldBool::displayInput(&$layout_def) should be compatible with SugarWidgetField::displayInput($layout_def) occurred in /var/www/crm.fortecait.pl/include/generic/SugarWidgets/SugarWidgetFieldbool.php on line 120 [2018-02-02 09:05:21] display_stack_trace caller, file: /var/www/crm.fortecait.pl/include/utils.php line#: 3517
/var/www/crm.fortecait.pl/include/utils/autoloader.php[L:93](:StackTraceErrorHandler)
/var/www/crm.fortecait.pl/include/utils/autoloader.php[L:93](:require_once)
[L:](SugarAutoLoader:autoload)
[L:](:spl_autoload_call)
/var/www/crm.fortecait.pl/include/generic/LayoutManager.php[L:310](:class_exists)
/var/www/crm.fortecait.pl/include/Dashlets/DashletGeneric.php[L:306](LayoutManager:getClassFromWidgetDef)
/var/www/crm.fortecait.pl/include/Dashlets/DashletGeneric.php[L:391](DashletGeneric:buildWhere)
/var/www/crm.fortecait.pl/modules/Opportunities/Dashlets/MyOpportunitiesDashlet/MyOpportunitiesDashlet.php[L:87](DashletGeneric:process)
/var/www/crm.fortecait.pl/modules/Home/index.php[L:236](MyOpportunitiesDashlet:process)
/var/www/crm.fortecait.pl/modules/Home/views/view.list.php[L:52](:include)
/var/www/crm.fortecait.pl/include/MVC/View/SugarView.php[L:199](HomeViewList:display)
/var/www/crm.fortecait.pl/include/MVC/Controller/SugarController.php[L:432](SugarView:process)
/var/www/crm.fortecait.pl/include/MVC/Controller/SugarController.php[L:375](SugarController:processView)
/var/www/crm.fortecait.pl/include/MVC/SugarApplication.php[L:105](SugarController:execute)
/var/www/crm.fortecait.pl/index.php[L:52](SugarApplication:execute)
Unknown Error (8192): Methods with the same name as their class will not be constructors in a future version of PHP; CalendarDashlet has a deprecated constructor occurred in /var/www/crm.fortecait.pl/modules/Calendar/Dashlets/CalendarDashlet/CalendarDashlet.php on line 45 [2018-02-02 09:05:21] display_stack_trace caller, file: /var/www/crm.fortecait.pl/include/utils.php line#: 3517
/var/www/crm.fortecait.pl/modules/Home/index.php[L:215](:StackTraceErrorHandler)
/var/www/crm.fortecait.pl/modules/Home/index.php[L:215](:require_once)
/var/www/crm.fortecait.pl/modules/Home/views/view.list.php[L:52](:include)
/var/www/crm.fortecait.pl/include/MVC/View/SugarView.php[L:199](HomeViewList:display)
/var/www/crm.fortecait.pl/include/MVC/Controller/SugarController.php[L:432](SugarView:process)
/var/www/crm.fortecait.pl/include/MVC/Controller/SugarController.php[L:375](SugarController:processView)
/var/www/crm.fortecait.pl/include/MVC/SugarApplication.php[L:105](SugarController:execute)
/var/www/crm.fortecait.pl/index.php[L:52](SugarApplication:execute)

**Comment by e-reeley** [comment]
@AdminPL Im sorry for the delay, I also still can't seem to replicate this issue. We can maybe wait to see if any others have an issue with this?

I have added in the custom fields as you did, set the conditions and run the same build.

Unless there is maybe something else that has been changed to the out of the box Suite?

Thanks again.

**Comment by AdminPL** [comment]
I have fiew custom modules but not connected with this one so i don't expect any problems. Installation was upgraded since version 7.6.5 and still sometimes i got problems with CHMOD, and have to change manually. 

I can try creating VM image copy so you could test it. 
Do you think it can be usefull ?

**Comment by PedroErnst** [comment]
Hello @AdminPL 

I've tried to reproduce the steps you mentioned, and no error occurs.

Report Fields

![screenshot from 2018-02-05 12-17-29](https://user-images.githubusercontent.com/34056696/35804194-a15e3e3c-0a6e-11e8-980a-47c072d24a5f.png)

Conditions

![screenshot from 2018-02-05 12-17-35](https://user-images.githubusercontent.com/34056696/35804199-a5c61788-0a6e-11e8-89ba-97d4d0c459da.png)

Result

![screenshot from 2018-02-05 12-17-46](https://user-images.githubusercontent.com/34056696/35804202-aa414ca6-0a6e-11e8-9cc8-cc5fcbc30b67.png)

Maybe some of the customizations that you think are not connected are in fact interfering with the report?

Or maybe there are some steps that you have done differently?

Either way, if we cannot reproduce this issue on a clean SuiteCRM installation we cannot fix it (because it is not a bug), taking a copy of your system setup is way beyond what we can reasonably do here.

**Comment by AdminPL** [comment]
I found what you do different - when i do it like youdo everything works fine, but when you make this condition to editable form main view report brokes.

![35804199-a5c61788-0a6e-11e8-89ba-97d4d0c459da](https://user-images.githubusercontent.com/30261902/35805320-a2aeb7aa-0a7b-11e8-9d2b-c1e59ed859ab.png)

**Comment by PedroErnst** [comment]
Interesting. However it still doesn't break it on my side...

![screenshot from 2018-02-05 13-01-50](https://user-images.githubusercontent.com/34056696/35805728-e12ec652-0a74-11e8-9706-a044086574c0.png)

**Comment by AdminPL** [comment]
only date parameter brakes report.

**Comment by PedroErnst** [comment]
Still working here.

![screenshot from 2018-02-05 13-06-39](https://user-images.githubusercontent.com/34056696/35805882-7477e178-0a75-11e8-8a7e-64c8e58a22b4.png)

**Comment by PedroErnst** [comment]
What kind of field is your custom date? Date or Datetime?

**Comment by AdminPL** [comment]
Try selecting range date equal to

**Comment by AdminPL** [comment]
also don't set any deault value, save report so it's empty and then select not from edit view to equal to range - for example this month.

**Comment by AdminPL** [comment]
Instead picking range date i added this condition 2 times and set one to be greater or equal to and the other to less or equal and i found interesting 0 added to date

![image](https://user-images.githubusercontent.com/30261902/35806053-728c4f6c-0a7e-11e8-84fd-75d67708858f.png)

**Comment by AdminPL** [comment]
PedroErnst - date

**Comment by PedroErnst** [comment]
That's interesting and might the cause of the bug, but I don't really understand your instructions. 

Please provide more specific, step for step details, including every little detail.

You give details only when requested, and now they're spread across a bunch of different posts and very hard to follow.

Ideally you would have provided these days ago and we would have solved this issue earlier, avoiding all of this back and forth which wastes a lot of time.

If for some reason you cannot provide steps then go with screenshots. Like the ones I sent, where all of the configurations of the report are clearly visible.

**Comment by AdminPL** [comment]
PedroErnest I'll make a video so you see every steps and piecie of configuration i got. Hope that will help.

**Comment by AdminPL** [comment]
https://youtu.be/jEi6WQYLUWk

In this video i recreated both scenario of how i'm trying to create report. 

1 scenario
Date range is created by 2 conditions 
- greater or equal to
- less or equal to

Report is empty some fields in report setting gone.

Whats interesting is when i "hardcode" paramters to report the first time i opne it data is complete and report works good, but editting any of date create empty report.


2 scenario
Date range is set by period in one field ex. one week - when i hit report to create page go blank.

**Comment by PedroErnst** [comment]
@AdminPL 

Ok, I've been able to reproduce at least parts of the issue now, and some are pretty straightforward fixes, which I will be putting forward soon.

In the future, let us agree that you start by providing all of these details from the very moment that you report the issue (video explanations are fine). You do some things in the video that you never really explained in writing, and that makes our life harder than it has to, and fixes for you slower than they need to be.

Thanks for your time!

**Comment by AdminPL** [comment]
Thanks for your help i'll in ther future i'll try to add as much detailed information as i can

**Comment by wayneoutthere** [comment]
hey guys.  i just wanted to leave a quick note here at the end for anyone reading this thread that i resolved my blank reports by something very, very simple: I had deleted a field via Studio.  That field, though, was still being asked for in the 'field' column of the report (I feel like this might be a bug...).  So, when I ran the report, all the conditions were right but the report wouldn't display because it could not find the missing field to put into the column. I hope this helps someone and I'll try to report that other bug somewhere...
